src/commands/aksDeployManifest/aksDeployManifest.ts

```diff
@@ -0,0 +1,145 @@
+    // Check if GitHub Copilot for Azure extension is installed
+    const checkGitHubCopilotExtension = checkExtension(GITHUBCOPILOT_FOR_AZURE_VSCODE_ID);
+
+    if(!checkGitHubCopilotExtension) {
+        handleExtensionDoesNotExist(GITHUBCOPILOT_FOR_AZURE_VSCODE_ID);
+        return;
+    }
+
+    // Select manifest
+    const manifest = await getManifestFile();
+
+    if(manifest === undefined) { 
+        return;
+    }
+
+    // Select cluster
+    const sessionProvider = await getReadySessionProvider();
+
+    if(failed(sessionProvider)) {
+        vscode.window.showErrorMessage(sessionProvider.error);
+        return;
+    }
+
+    const cluster = await selectClusterOptions(sessionProvider.result);
+
+    if (failed(cluster)) {
+        vscode.window.showErrorMessage(cluster.error);
+        return;
+    }
+
+    if (cluster.result === false) {
+        vscode.window.showWarningMessage("No cluster selected.");
+        return;
+    }
+
+    const clusterPreference = cluster.result as ClusterPreference;
+
+    // Confirm deployment
+    const confirmed = await confirmDeployment(clusterPreference.clusterName);
+
+    if(!confirmed) {
+        vscode.window.showWarningMessage("Deployment cancelled.");
+        return;
+    }
+
+    const deploymentResult = await deployApplicationToCluster({
+        cluster: clusterPreference,
+        manifestPath: manifest
+    });
+
+    if(failed(deploymentResult)) {
+        vscode.window.showErrorMessage(deploymentResult.error);
+        return;
+    }
+
+    vscode.window.showInformationMessage(`Your applicatoin has been successfully deployed to the ${clusterPreference.clusterName} AKS cluster.`, "View resource in the Azure portal").then((res) => {
+        if(res) {
+            vscode.env.openExternal(vscode.Uri.parse(deploymentResult.result.url));
+        }
+    });
+}
+
+type ManifestQuickPickItem = vscode.QuickPickItem & {filePath: string};
+
+async function getManifestFile(): Promise<string | undefined> {
+    const items: ManifestQuickPickItem[] = [];
+    await vscode.workspace.findFiles(`**/**.yaml`, "**/node_modules/**").then((result) => {
+        result.forEach((fileUri) => {
+            const fileName = path.basename(fileUri.fsPath);
+            items.push({ label: fileName, description: fileUri.fsPath, filePath: fileUri.fsPath });
+        });
+    });
+
+    if(items.length === 0) {
+        vscode.window.showWarningMessage("No manifest files found in the workspace.");
+        return;
+    }
+
+    const fileSelected = await vscode.window.showQuickPick(items.sort(), {
+        title: "Select YAML",
+        placeHolder: "Select manifest to deploy",
+    });
+
+    if (!fileSelected) {
+        vscode.window.showWarningMessage("Manifest file not selected");
+        return;
+    }
+
+    return fileSelected.filePath;
+}
+
+async function confirmDeployment(clusterName: string): Promise<boolean> {
+    const confirmed = await vscode.window.showQuickPick(["Yes", "No"], {
```

Comment:
💡 Nit: purely for refactor sake latter, aka lets handle latter if you want: at this point it is not `confirmed` I think it still a selection, but its file its a nomenclature for code readability. Thanks heaps.

Reply:
Thanks, renamed to "selection" to make it more clear